### PR TITLE
Forms component

### PR DESCRIPTION
**Pull request by MFarmer**
Adds a Form Widget to the Widget Builder.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,5 @@
 /bower_components
 /manifest.json
 /node_modules
+.idea/
+*.swp
```
Comment (caseywebdev) [inline .gitignore]: I would recommend adding these to your global `.gitignore`. That way they don't have to be added to every `.gitignore` in every project you work on.

**Comment by caseywebdev** [comment]
Looks like you checked in `scripts/components/.summary.es6.swp` on accident.

File: scripts/components/shared/category-selector.es6 [hunk 2]
```diff
@@ -1,21 +1,27 @@
...
+import FacetedSelector from 'components/shared/faceted-selector';
 import joinClassNames from 'utils/join-class-names';
 import React from 'react';
 
 export default React.createClass({
   mixins: [Cursors],
 
-  getFacet: function (portal) { return portal.category.name; },
+  propTypes: {
+    objects: React.PropTypes.array,
```
Comment (caseywebdev) [inline scripts/components/shared/category-selector.es6]: Is there a case where this isn't required?

**Comment by caseywebdev** [comment]
Couple comments but LGTM

File: scripts/components/forms/index.es6 [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    );
+  },
+
+  renderLoading: function () {
+    return <div className='osw-inset-block'>Loading...</div>;
+  },
+
+  renderError: function (er) {
+    return (
+      <div className='osw-inset-block osw-inset-block-red'>{er.toString()}</div>
```
Comment (clifton) [inline scripts/components/forms/index.es6]: Kinda seems like this DOM + classes for the error div and for the loading div should be abstracted out, but that's another project for another day.

Comment (caseywebdev) [inline scripts/components/forms/index.es6]: Yea that kind of stuff can be abstracted into components all over the widgets.

Comment (tylerlee) [inline scripts/components/forms/index.es6]: and documented.

Comment (clifton) [inline scripts/components/forms/index.es6]: Yeah, seems like any classes or DOM structure that's global in nature should be in a global shared component set. All the other DOM/classes in this pull req are specific to the forms module like it should be.

Comment (caseywebdev) [inline scripts/components/forms/index.es6]: That's what the `scripts/components/ui` folder is for

**Pull request merged**